File: bindgen-tests/swift/tests/complex_fns.swift
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import uniffi_bindgen_tests

// Default arguments
assert(funcWithDefault() == "DEFAULT");
assert(funcWithDefault(arg: "NON-DEFAULT") == "NON-DEFAULT");

let complexMethods = ComplexMethods();
assert(complexMethods.methodWithDefault() == "DEFAULT")
assert(complexMethods.methodWithDefault(arg: "NON-DEFAULT") == "NON-DEFAULT")


// These just test that the argument names get mapped to camelCase
let _ = complexMethods.methodWithMultiWordArg(theArgument: "test")
let _ = funcWithMultiWordArg(theArgument: "test")

File: bindgen-tests/python/tests/complex_fns.py
from test_package.uniffi_bindgen_tests import *
import unittest

class TestComplexFns(unittest.TestCase):
    def test_default_arguments(self):
        self.assertEqual(func_with_default(), "DEFAULT")
        self.assertEqual(func_with_default("NON-DEFAULT"), "NON-DEFAULT")

        complex_methods = ComplexMethods()
        self.assertEqual(complex_methods.method_with_default(), "DEFAULT")
        self.assertEqual(complex_methods.method_with_default("NON-DEFAULT"), "NON-DEFAULT")

    def test_argument_name_mapping(self):
        complex_methods = ComplexMethods()
        # the following calls will fail if the argument name differs
        func_with_multi_word_arg(the_argument="test")
        complex_methods.method_with_multi_word_arg(the_argument="test")

if __name__ == '__main__':
    unittest.main()
